Source/WebCore/Modules/applepay/PaymentInstallmentConfiguration.mm

```diff
@@ -194,72 +200,72 @@ static PKInstallmentRetailChannel platformRetailChannel(ApplePayInstallmentRetai
         return configuration;
 
     [configuration setInstallmentItems:createNSArray(coreConfiguration.items).get()];
-    [configuration setApplicationMetadata:applicationMetadata];
+    [configuration setApplicationMetadata:applicationMetadataDictionary(coreConfiguration).get()];
     [configuration setRetailChannel:platformRetailChannel(coreConfiguration.retailChannel)];
 
     return configuration;
 }
 
 ExceptionOr<PaymentInstallmentConfiguration> PaymentInstallmentConfiguration::create(const ApplePayInstallmentConfiguration& configuration)
 {
-    NSDictionary *applicationMetadataDictionary = nil;
-    if (!configuration.applicationMetadata.isNull()) {
-        NSData *applicationMetadata = [configuration.applicationMetadata dataUsingEncoding:NSUTF8StringEncoding];
-        applicationMetadataDictionary = dynamic_objc_cast<NSDictionary>([NSJSONSerialization JSONObjectWithData:applicationMetadata options:0 error:nil]);
-        if (!applicationMetadataDictionary)
-            return Exception { TypeError, "applicationMetadata must be a JSON object"_s };
-    }
+    if (!configuration.applicationMetadata.isNull() && !applicationMetadataDictionary(configuration))
```

Comment:
Doesn't seem great to call `applicationMetadataDictionary` twice: once as a null-check then again in the `PaymentInstallmentConfiguration` ctor. Can we pass the metadata dictionary as an argument to the ctor?